### ISSUE TITLE
UserDistanceValueDriver<Material> prevents world saving

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#1226]
Steps to replicate:
- Attach UserDistanceValueDriver of type Material to a slot
- Attempt to save the world

The world will not, and can not be saved, and the component can not be used or removed.
```
8:19:25 AM.375 (364 FPS)	Saving world: UserDistanceValueDriver<Material> Bug, currently being saved: 1
8:19:25 AM.406 (364 FPS)	World Optimized! Deduplicated Materials: 0, Deduplicated Static Providers: 7, Cleaned Up Assets: 18
8:19:25 AM.642 (364 FPS)	Exception when running synchronous action (Stage: UpdatingStreams):
System.NullReferenceException: Object reference not set to an instance of an object
  at FrooxEngine.Coder`1[T].Save (T value) [0x00000] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.Sync`1[T].InternalSave (FrooxEngine.SaveControl control) [0x00006] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.SyncElement.Save (FrooxEngine.SaveControl control) [0x0001c] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.Worker.Save (FrooxEngine.SaveControl control) [0x0008b] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.WorkerSaveLoad.SaveWorker (FrooxEngine.IWorker worker, FrooxEngine.SaveControl control) [0x0001b] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.WorkerBag`1[W].InternalSave (FrooxEngine.SaveControl control) [0x0009b] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.SyncElement.Save (FrooxEngine.SaveControl control) [0x0001c] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.Worker.Save (FrooxEngine.SaveControl control) [0x0008b] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.Slot.Save (FrooxEngine.SaveControl control) [0x0001d] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.Slot.Save (FrooxEngine.SaveControl control) [0x0009a] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.World.Save (FrooxEngine.SaveControl control) [0x000fe] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.World.SaveWorld () [0x00035] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.Userspace+<>c__DisplayClass207_0.<SaveWorldTaskIntern>b__0 () [0x00048] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
  at FrooxEngine.World.RunSynchronousActions () [0x0007b] in <bb9c5bb13f124ca2a9968cc9f9be221d>:0 
```

Issue discovered, and reported by Jeff07734# 5022 and reddragon# 1170.
[SHIFTYWINDOWS - 2020.10.28.1149 - 2020-10-29 08_16_45.log](https://github.com/Frooxius/NeosPublic/files/5459457/SHIFTYWINDOWS.-.2020.10.28.1149.-.2020-10-29.08_16_45.log)

**Comment by Frooxius** [comment]
This is because Material is invalid generic argument for this component. I'll add a check to make sure you can't create this component with this type, it's for Value types only.

**Comment by Frooxius** [comment]
Fixed in 2020.10.29.1313! I've also modified the save routine, so errors like this will cause the offending component to be skipped, but won't prevent saving altogether.
